doc-content/shared-kie-docs/src/main/asciidoc/KieServer/kie-server-rest-api-solvers-ref.adoc

```diff
@@ -0,0 +1,419 @@
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solvers reside
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "solver": [
+    {
+      "container-id": "employee-rostering",
+      "solver-id": "solver1",
+      "solver-config-file": "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml",
+      "status": "NOT_SOLVING"
+    }
+  ]
+}
+----
+
+== [PUT] /containers/{containerId}/solvers/{solverId}
+
+Creates a solver within a specified KIE container.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solver is to be placed
+|String
+|Required
+
+|`solverId`
+|ID of the new solver
+|String
+|Required
+
+|*body*
+|A `solver-config-file` specification (for XML requests, a marshalled `SolverInstance` entity)
+|Request body
+|Required
+|===
+
+.Example request body (JSON)
+[source,json]
+----
+{
+  "solver-config-file": "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml"
+}
+----
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "solver-instance": {
+    "container-id": "employee-rostering",
+    "solver-id": "solver1",
+    "solver-config-file": "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml",
+    "status": "NOT_SOLVING",
+    "score": {
+      "value": null,
+      "scoreClass": null
+    },
+    "best-solution": null
+  }
+}
+----
+
+== [GET] /containers/{containerId}/solvers/{solverId}
+
+Returns the current state of the specified solver within the specified KIE container.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solver resides
+|String
+|Required
+
+|`solverId`
+|ID of the solver
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "solver-instance": {
+    "container-id": "employee-rostering",
+    "solver-id": "solver1",
+    "solver-config-file": "employeerostering/employeerostering/EmployeeRosteringSolverConfig.solver.xml",
+    "status": "NOT_SOLVING"
+  }
+}
+----
+
+== [DELETE] /containers/{containerId}/solvers/{solverId}
+
+⁠Disposes the specified solver within the specified KIE container. If the solver is still running, this request terminates the solver and then disposes it.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`containerId`
+|ID of the KIE container where the solver resides
+|String
+|Required
+
+|`solverId`
+|ID of the solver
+|String
+|Required
+|===
+
+*<@Jiri: The request works but the server responds "undocumented". This example JSON response body below is what the server needs to say.>*
```

Comment:
Not sure if that's the expected response. Let's see how it works with other endpoints (but those are currently blocked by https://issues.jboss.org/browse/PLANNER-1356).